spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -501,34 +498,29 @@ function onChanCloseConfirm(
 
 ```typescript
 function OnRecvPacket(packet Packet) {
-    	ack := NewResultAcknowledgement([]byte{byte(1)})
+    	ack = NewResultAcknowledgement([]byte{byte(1)})
 
 	// only attempt the application logic if the packet data
 	// was successfully decoded
-	if ack.Success() {
-		switch data.Type {
-	          case types.EXECUTE_TX:
-		        msgs, err := types.DeserializeTx(data.Data)
-		        if err != nil {
-			      return err
-		        }
-
-                        // ExecuteTx calls the AuthenticateTx function defined above 
-		        if err = ExecuteTx(ctx, packet.SourcePort, packet.DestinationPort, packet.DestinationChannel, msgs); err != nil {
-			      return err
-		        }
-
-		        return nil
-	          default:
-		        return ErrUnknownDataType
-	        }
-	        if err != nil {
-                    ack = NewErrorAcknowledgement(err.Error())
-                }
-	}
-
-	// NOTE: acknowledgment will be written synchronously during IBC handler execution.
-	return ack
+  switch data.Type {
+  case types.EXECUTE_TX:
+  msgs, err := types.DeserializeTx(data.Data)
+  if err != nil {
+    return NewErrorAcknowledgement(err)
+  }
+
+  // ExecuteTx calls the AuthenticateTx function defined above 
+  result, err = ExecuteTx(ctx, packet.SourcePort, packet.DestinationPort, packet.DestinationChannel, msgs)
+  if err != nil {
+    return NewErrorAcknowledgement(err)
+  }
+
+  // return acknowledgement containing the transaction result after executing on host chain
+  return NewAcknowledgement(result)
```

Comment:
ditto for implications of including the result

Reply:
The result of a transaction is currently hashed and included in the the Tendermint block header (DataHash I believe). The data field and the code (u32) are the only fields currently consensus breaking (not events). I think including Data is fully acceptable, where errors and events may need more consideration.

Reply:
Great insight! I did not know the data field is included in consensus. I agree, if the result is already included in consensus then it is fully acceptable to include in the acknowledgement